functions/src/SQR.functions.js

```diff
@@ -0,0 +1,234 @@
+//          3.2 the data of the file in the submission
+//          3.3 the list of all the files alloted to the devotee of the current submission
+//          3.4 a boolean value indicating whether this is the first submission of this devotee or not
+//              Function --> processSubmissions
+/////////////////////////////////////////////////
+
+
+exports.processSubmissions = functions.database.ref('/webforms/sqr/{submission_id}')
+    .onCreate((snapshot, context) => {
+        const original = snapshot.val();
+
+        let audioFileStatus = 'WIP';
+        
+
+        if(original.not_preferred_language) 
+            audioFileStatus = 'Spare';        
+        else if(original.unable_to_play_or_download)
+            audioFileStatus = 'Audio Problem';
+        
+
+        // 1. Add the webform data to a SQR submissions DB path
+        let submission = {
+            fileName: original.audio_file_name,
+            cancellation: {
+                notPreferredLanguage: audioFileStatus === 'Spare',
+                audioProblem: audioFileStatus === 'Audio Problem'
+            },
+            soundQualityRating: original.sound_quality_rating,
+            unwantedParts: original.unwanted_parts,
+            soundIssues: original.sound_issues,
+            duration: {
+                beginning: original.beginning,
+                ending: original.ending,
+            },
+            comments: original.comments,
+            token: original.token,
+            created: original.created, 
+            //  timestamp of the submission creation,
+            // can differ from COMPLETED in case of saving a DRAFT and completing later.
+            completed: original.completed, // timestamp of the submission completion.
+            changed: original.changed, //timestamp of the submission update.
+            devotee: {
+                name: original.name,
+                emailAddress: original.email_address
+            }
+        };
+        db.ref(`/sqr/submissions/${original.serial}`).update(submission);
+
+
+
+        // 2. Update the allotment
+        let allotmentUpdates = { status: audioFileStatus };
+
+        // in case 1 & 2 add the comments to the notes
+        if (audioFileStatus !== 'WIP')
+            allotmentUpdates.notes = original.comments;
+        
+        // if the audio has a problem then REMOVE the devotee from the file allotment
+        if(audioFileStatus === 'audioProblem')
+            allotmentUpdates.devotee = {};
+
+        db.ref(`/sqr/files/${original.list}/${original.audio_file_name}`).update(allotmentUpdates);
+
+
+
+
+        // Coordinator object example { templateid: 3, email:'a@a.a', name: 'Aj' }
+        let coordinator = functions.config().audioseva.coordinator;
+        // 3. Notify the coordinator
+        // 3.1 Get the submitted audio file data
+
+        //  EXTRACTING the list name first from the file_name
+        let list = original.audio_file_name.match(/^[^-]*[^ -]/g)[0];
```

Comment:
This is a good candidate for a helper, as you will be extracting list many times from various places.